controllers/util/onboarding_secrets.go

```diff
@@ -0,0 +1,41 @@
+package util
+
+import (
+	"context"
+	"crypto/rsa"
+	"crypto/x509"
+	"encoding/pem"
+	"fmt"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/klog/v2"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func ReadPrivateKey(cl client.Client) (*rsa.PrivateKey, error) {
+	klog.Info("Getting the Pem key")
+	ctx := context.Background()
+
+	operatorNamespace, err := GetOperatorNamespace()
```

Comment:
get the namespace as param rather than reading it everytime, I see this is being used in two places where you can already have ns.
1. in reconciler, already has it
2. in handler, pls initialize ns once and read from it rather than reading from env everytime, ref #2569

Reply:
ack